packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx

```diff
@@ -31,586 +32,566 @@ const behaviorMenu: ComponentMenuItem[] = require('../../behaviorMenu');
 
-    if (this.state.query !== '') {
-      allSections = _.map(allSections, (section: HierarchicalTreeItemProps) => {
-        return { ...section, open: true };
-      });
-    }
+    const at = props.location.pathname;
+    const id = findActiveCategoryId(at, treeItems);
+    setActiveItemIds(prev => (prev.includes(id) ? prev : [...prev, id]));
+    return () => {
+      document.removeEventListener('keydown', handleDocumentKeyDown);
+    };
+  }, [handleDocumentKeyDown, props.location.pathname, treeItems]);
 
-    // TODO: remove after the issue with TreeItem will be fixed
-    // https://github.com/microsoft/fluent-ui-react/issues/1613
-    this.addItemKeyCallbacks(allSections);
+  const keyDownCallback = e => {
+    if (getCode(e) !== keyboardKey.Enter) {
+      return;
+    }
+    e.stopPropagation();
+    e.target.click();
+  };
 
-    this.addItemOnClickCallbacks(allSections);
+  const addItemKeyCallbacks = React.useCallback((sections: ShorthandValue<any>[]) => {
+    for (let i = 0; i < sections.length; i++) {
+      const category = sections[i];
+      if ('items' in category) {
+        addItemKeyCallbacks(category.items);
+      } else {
+        if (!('title' in category)) {
+          continue;
+        }
+        category['onKeyDown'] = e => {
+          keyDownCallback(e);
+        };
+      }
+    }
+  }, []);
+
+  const allSections = React.useMemo(
+    () =>
+      treeItems
+        .map((section: TreeItemProps) => {
+          return {
+            ...section,
+            items: _.filter(section.items as TreeItemProps[], item =>
+              regexQuery.test((item.title as TreeTitleProps).content as string),
+            ),
+          };
+        })
+        .filter((section: TreeItemProps) => Array.isArray(section.items) && section.items.length > 0),
+    [regexQuery, treeItems],
+  );
+
+  const handleItemClick = React.useCallback(
+    (e: React.SyntheticEvent, data: TreeItemProps) => {
+      if (query) {
+        setQuery('');
+        const at = (data.title as NavLinkProps).to as string;
+        const id = findActiveCategoryId(at, treeItems);
+        setActiveItemIds(prev => (prev.includes(id) ? prev : [...prev, id]));
+      }
+    },
+    [query, treeItems],
+  );
+
+  const addItemOnClickCallbacks = React.useCallback(
+    (sections: ShorthandValue<any>[]) => {
+      for (let i = 0; i < sections.length; i++) {
+        const category = sections[i];
+        if ('items' in category) {
+          addItemOnClickCallbacks(category.items);
+        } else {
+          if (!('title' in category)) {
+            continue;
+          }
+          category['onTitleClick'] = (e, data) => {
+            handleItemClick(e, data);
+          };
+        }
+      }
+    },
+    [handleItemClick],
+  );
+
+  React.useEffect(() => {
+    addItemKeyCallbacks(treeItems);
+    addItemOnClickCallbacks(treeItems);
+  }, [addItemKeyCallbacks, addItemOnClickCallbacks, treeItems]);
```

Comment:
Why this defined as `useEffect`?

Reply:
Removed this one, it was problem with propagation on the `Tree`